### ISSUE TITLE
provider_settings validation fails when respository is dynamic

### ISSUE THREAD

**Issue by jderusse** [buildkite/terraform-provider-buildkite#409]
**Describe the bug**
When using the resource `buildkite_pipeline` in a module with a repository containing variable interpolation, the validation of the `provider_settings` fails

**To Reproduce**

A small reproducer
```
resource "buildkite_pipeline" "main" {
  name        = var.name
  repository  = "git@github.com:${var.repository}.git"

  provider_settings = {
    trigger_mode = "code"
  }
}
```

Trigger this output 
```
╷
│ Error: Could not parse repository URL
│ 
│   with module.agent_release.buildkite_pipeline.main,
│   on modules/buildkite_release/main.tf line 3, in resource "buildkite_pipeline" "main":
│    3: resource "buildkite_pipeline" "main" {
│ 
╵
╷
│ Error: Invalid attribute value
│ 
│   with module.agent_release.buildkite_pipeline.main,
│   on modules/buildkite_release/main.tf line 3, in resource "buildkite_pipeline" "main":
│    3: resource "buildkite_pipeline" "main" {
│ 
│ Invalid use when repository is: 
╵
```

**Additional context**

terraform-provider-buildkite Version 1.0

Probably introduced in https://github.com/buildkite/terraform-provider-buildkite/pull/387/files

**Comment by jradtilbrook** [comment]
Hmm damn. That is an edge that wasn't considered or tested. Thanks for reporting @jderusse 👍

It seems that validation runs prior to interpolation, even with known variables. I think we could fix this by implementing a plan modifier instead or perhaps we ignore the validation when the string is empty. But that would mean there is no validation when interpolating the repository URL which I suspect would be how many people use this provider. Let us have a think about which approach and implement a fix

**Comment by ianchesal** [comment]
I am also encountering this problem. In my case my module is:

```
resource "buildkite_pipeline" "pipeline" {
  # See: https://registry.terraform.io/providers/buildkite/buildkite/latest/docs/resources/pipeline
  for_each = local.pipelines

  # Required settings
  default_branch = each.value.default_branch
  description    = each.value.description
  name           = each.key
  repository     = each.value.repository
  ...snip...
```

And from this I'm getting:

```
│ Error: Invalid attribute value
│
│   with buildkite_pipeline.pipeline["persona-identity"],
│   on main.tf line 1, in resource "buildkite_pipeline" "pipeline":
│    1: resource "buildkite_pipeline" "pipeline" {
│
│ Invalid use when repository is: git@bitbucket.org:persona-id/persona-identity.git
```

I'm also seeing similar issues using variables for provider settings. I have:

```
  provider_settings = {
   ...snip...
    pull_request_branch_filter_configuration = lookup(each.value, "pull_request_branch_filter_configuration", "")
    pull_request_branch_filter_enabled = lookup(each.value, "pull_request_branch_filter_enabled", false)
    ...snip...
  }
```

And as a result I'm geting plans failing with:

```
│ Error: Invalid Attribute Value
│
│   with buildkite_pipeline.pipeline["persona-identity"],
│   on main.tf line 1, in resource "buildkite_pipeline" "pipeline":
│    1: resource "buildkite_pipeline" "pipeline" {
│
│ Attribute provider_settings.pull_request_branch_filter_enabled true, got: false
```

Note: `false` is actually what the value should be here as it's not set explicitly in the instantiated instance of the module.

**Comment by jradtilbrook** [comment]
Thanks @ianchesal. This looks semi-related, but distinct enough that I'd like to ask you raise a separate issue for us (I think it could be a bit more complex). Could you also provide a reproduction example for us to figure out the issue

**Comment by nickburgin** [comment]
This got us too. Simply using a variable for repository leads to validation failures.

In our simpler module, we just create a pipeline from the variables directly
```
resource "buildkite_pipeline" "buildkite_pipeline" {
  name           = var.service_name
  repository     = var.repository
  description    = var.description
  steps          = var.steps
  default_branch = var.default_branch

  provider_settings = {
    build_branches                                = true
    build_pull_requests                           = true
    publish_commit_status                         = true
    publish_commit_status_per_step                = true
    skip_pull_request_builds_for_existing_commits = true
    trigger_mode                                  = var.trigger_mode
  }
}
```

```
variable "repository" {
  type = string
}
...
```

Our more complex module creates a pipeline as a side effect of creating a GitHub repo, this also fails:
```
resource "buildkite_pipeline" "buildkite_pipeline" {
  count          = var.build_pipeline ? 1 : 0
  name           = github_repository.github_repository.name
  repository     = github_repository.github_repository.ssh_clone_url
  steps          = var.buildkite_steps
  default_branch = (var.build_default_branch != null) ? var.build_default_branch : var.repo_default_branch

  provider_settings = {
    build_branches                                = true
    build_pull_requests                           = true
    publish_commit_status                         = true
    publish_commit_status_per_step                = true
    skip_pull_request_builds_for_existing_commits = true
    trigger_mode                                  = "code"
  }
}
```

**Comment by jradtilbrook** [comment]
Hi all, we've released v1.0.2 with a fix for this. Can you give it a shot and let us know if it still doesn't work or raise a new issue if some other bug comes up

**Comment by dotnich-io** [comment]
@jradtilbrook I encountered this today, trying to create a module that generated a GH repository and an associated pipeline. This was with version `1.3.0`
